compress is failing code quality checks

In d8984cd37f62a220ba784e634a3328b47c19c64f a change was introduced which causes `make check` to fail and thus the build.

https://github.com/rclone/rclone/blob/0be69018b8498f2680263ccdbbf183e92012caf9/backend/compress/compress.go#L1224-L1236

This is failing the build with

```
$ make check
go build -v --ldflags "-s -X github.com/rclone/rclone/fs.Version=v1.54.0-beta.5072.0be69018b"  
mkdir -p `go env GOPATH`/bin/
cp -av rclone`go env GOEXE` `go env GOPATH`/bin/rclone`go env GOEXE`.new
'rclone' -> '/home/ncw/go/bin/rclone.new'
mv -v `go env GOPATH`/bin/rclone`go env GOEXE`.new `go env GOPATH`/bin/rclone`go env GOEXE`
renamed '/home/ncw/go/bin/rclone.new' -> '/home/ncw/go/bin/rclone'
-- START CODE QUALITY REPORT -------------------------------
backend/compress/compress.go:1227:6: ineffectual assignment to `ok` (ineffassign)
	do, ok := o.Object.(fs.SetTierer)
	    ^
make: *** [Makefile:87: check] Error 1
```

I can see the problem, `ok` is being assigned but not checked, so the `return do.SetTier(tier)` call might panic.

I'm not sure what the solution is though - should it be looking at that object at all?

Can you fix it @buengese ?

--- a/backend/compress/compress.go
+++ b/backend/compress/compress.go
@@ -1225,8 +1225,8 @@ func (o *Object) Hash(ctx context.Context, ht hash.Type) (string, error) {
 // multiple storage classes supported
 func (o *Object) SetTier(tier string) error {
 	do, ok := o.Object.(fs.SetTierer)
-	mdo, ok := o.mo.(fs.SetTierer)
-	if !ok {
+	mdo, mok := o.mo.(fs.SetTierer)
+	if !(ok && mok) {
 		return errors.New("press: underlying remote does not support SetTier")
 	}
 	if err := mdo.SetTier(tier); err != nil {

